src/scripts/migrate_utils.py

```diff
@@ -0,0 +1,257 @@
+
+        # Set: uri, curie
+        prefix_map = general_prefix_map if general_prefix_map else owned_prefix_map
+        if prefix_map and self.uri and not self.curie:
+            converter = self._converter if self._converter else curies.Converter.from_prefix_map(prefix_map)
+            curie = converter.compress(self.uri)
+            self.curie = curie if curie != self.uri else None
+        elif prefix_map and self.curie and not self.uri:
+            converter = curies.Converter.from_prefix_map(prefix_map)
+            uri = converter.expand(self.curie)
+            self.uri = uri if uri != self.curie else None
+
+        # Set: label, definition
+        # todo: When OAK supports: I feel like this would be faster to do outside the Term class itself
+        #  ...e.g. build a dict of CURIE-> Term, then use use ontology.labels() on all curies and set label
+        #  ...Right now, OAK ProntoImplementation labels() is just proxy for label(), though SqlImplementation has, and
+        #  ...Both of them have definition() but not definitions().
+        if ontology:
+            self.label = ontology.label(self.curie)
+            self.definition = ontology.definition(self.curie)
+
+        # Set: direct_parent_uris
+        if ontology and owned_prefix_map:
+            self.direct_parent_uris = self._get_direct_owned_parents(
+                owned_prefix_map=owned_prefix_map if owned_prefix_map else self.owned_prefix_map)
+
+    def __repr__(self):
+        return f'<Term({self.curie if self.curie else self.uri})>'
+
+
+# TODO: implement this func
+# todo: IDs should be int or str? prolly str
+def _get_next_available_mondo_id(min_id: Union[str, int], mondo_ids: Set[int]) -> (int, Set[int]):
+    """Starting from `min_id`, count up and check until finding the next ID.
+
+    :returns: (int) next mondo ID; (Set[int]) New set of Mondo IDs, assuming 'next mondo ID' is included."""
+    min_id = int(min_id)
+    while True:
+        next_id = min_id + 1
+        if next_id not in mondo_ids:
+            break
+    mondo_ids = mondo_ids.add(next_id)
+    return next_id, mondo_ids
+
+
+def get_mondo_term_ids(path: str) -> Set[int]:
+    """From path to file of mondo terms, get set of Mondo IDs as integers.
+    # todo: Consider using `curies` package, though needing another prefix_map only for this seems even less optimal
+    """
+    mondo_base_uri = 'http://purl.obolibrary.org/obo/MONDO_'
+    mondo_termlist_df = pd.read_csv(path, comment='#', sep='\t')
+    mondo_term_uris: List[str] = list(mondo_termlist_df['?term'])
+    mondo_ids: List[int] = []
+    for x in mondo_term_uris:
+        x = x[1:] if x.startswith('<') else x
+        x = x[:-1] if x.endswith('>') else x
+        if not x.startswith(mondo_base_uri):
+            continue
+        mondo_ids.append(int(x.replace('http://purl.obolibrary.org/obo/MONDO_', '')))
+    return set(mondo_ids)
+
+
+def _get_direct_owned_parent_uris(
```

Comment:
FYI: `_get_direct_owned_parents()`: For a given URI, returns the direct parents of a class that are 'owned'b y the given ontology.

This code could probably be copy/pasted into OAK with minimal changes if we want to add this feature to OAK (I do).

Some things I would add to this:
- accept both CURIE or URI
- return a CURIE or URI; defaults to whatever type was passed as the lookup term